cmd/iam.go

```diff
@@ -71,21 +74,48 @@ func (sys *IAMSys) Init(objAPI ObjectLayer) error {
 		return errInvalidArgument
 	}
 
-	defer func() {
-		// Refresh IAMSys in background.
-		go func() {
-			ticker := time.NewTicker(globalRefreshIAMInterval)
-			defer ticker.Stop()
-			for {
-				select {
-				case <-GlobalServiceDoneCh:
-					return
-				case <-ticker.C:
-					sys.refresh(objAPI)
+	if globalEtcdClient != nil {
+		defer func() {
+			go func() {
+				// Refresh IAMSys with etcd watch.
+				ctx, cancel := context.WithTimeout(context.Background(), defaultContextTimeout)
+				defer cancel()
+				for {
+					select {
+					case <-GlobalServiceDoneCh:
+						return
+					case watchResp := <-globalEtcdClient.Watch(ctx, iamConfigPrefix):
+						if err := watchResp.Err(); err != nil {
+							logger.LogIf(ctx, err)
+							// log and retry.
```

Comment:
need to add a sleep before continuing - otherwise it hogs cpu if etcd is down

Reply:
Fixed